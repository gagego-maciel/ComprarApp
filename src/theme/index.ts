export const theme = {
  colors: {
    primary: '#8257E5',
    background: '#121214',
    backgroundWhite: '#ffffff',
    backgroundSuccess: '#27ae60',
    backgroundButtonBlue: '#2c46b1',
    backgroundButtonRed: '#E53E3E',
    text: '#E1E1E6',
    textBlack: '#000000',
    textWhite: '#ffffff',
    textGray: '#A0A0A0',

    border: '#c3c5cb',

    placeHolderInput: '#74798B',
  },
}
