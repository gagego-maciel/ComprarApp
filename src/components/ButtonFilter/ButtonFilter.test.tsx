import React from 'react'
import { ButtonFilter } from './index'
import '@testing-library/jest-native/extend-expect'
import { render, fireEvent } from '@testing-library/react-native'
import { ThemeProvider } from 'styled-components/native'
import { theme } from '@/theme'

const FILTER_CONFIG = {
  ALL: { label: 'Todos', iconActive: 'icon', iconDisabled: 'icon' },
  PENDING: { label: 'Pendentes', iconActive: 'icon', iconDisabled: 'icon' },
  PURCHASED: { label: 'Comprados', iconActive: 'icon', iconDisabled: 'icon' },
}

jest.mock('@/components/ButtonFilter/hooks/useButtonFilter', () => ({
  useButtonFilter: () => ({
    FILTER_CONFIG,
  }),
}))

const renderWithTheme = (ui: React.ReactElement) =>
  render(<ThemeProvider theme={theme}>{ui}</ThemeProvider>)

describe('ButtonFilter Component', () => {
  const statuses = ['ALL', 'PENDING', 'PURCHASED'] as const

  test('renders correct label and style depending on isActive', () => {
    statuses.forEach((status) => {
      const { getByText, unmount } = renderWithTheme(
        <ButtonFilter status={status} isActive={true} onPress={() => {}} />,
      )

      const label = getByText(FILTER_CONFIG[status].label)
      expect(label).toBeTruthy()
      expect(label).toHaveStyle({ color: theme.colors.textBlack })

      unmount()
    })

    statuses.forEach((status) => {
      const { getByText, unmount } = renderWithTheme(
        <ButtonFilter status={status} isActive={false} onPress={() => {}} />,
      )

      const label = getByText(FILTER_CONFIG[status].label)
      expect(label).toBeTruthy()
      expect(label).toHaveStyle({ color: theme.colors.textGray })

      unmount()
    })
  })

  test('calls onPress handler when pressed', () => {
    const onPressMock = jest.fn()

    statuses.forEach((status) => {
      const { getByTestId, unmount } = renderWithTheme(
        <ButtonFilter status={status} isActive={true} onPress={onPressMock} />,
      )

      const button = getByTestId(`button-${status}`)
      fireEvent.press(button)
      expect(onPressMock).toHaveBeenCalledTimes(1)

      onPressMock.mockClear()
      unmount()
    })
  })

  test('should match snapshot for all statuses', () => {
    const { toJSON } = renderWithTheme(
      <>
        <ButtonFilter status="ALL" isActive={true} onPress={() => {}} />
        <ButtonFilter status="ALL" isActive={false} onPress={() => {}} />
        <ButtonFilter status="PENDING" isActive={true} onPress={() => {}} />
        <ButtonFilter status="PENDING" isActive={false} onPress={() => {}} />
        <ButtonFilter status="PURCHASED" isActive={true} onPress={() => {}} />
        <ButtonFilter status="PURCHASED" isActive={false} onPress={() => {}} />
      </>,
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
