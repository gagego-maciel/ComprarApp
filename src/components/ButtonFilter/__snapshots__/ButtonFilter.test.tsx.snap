// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonFilter Component should match snapshot for all statuses 1`] = `
[
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 5,
        "height": 18,
        "maxWidth": 90,
        "opacity": 1,
      }
    }
    testID="button-ALL"
  >
    <Image
      source="icon"
      style={{}}
    />
    <Text
      isActive={true}
      style={
        {
          "color": "#000000",
          "fontSize": 12,
          "fontWeight": "500",
        }
      }
    >
      Todos
    </Text>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 5,
        "height": 18,
        "maxWidth": 90,
        "opacity": 1,
      }
    }
    testID="button-ALL"
  >
    <Image
      source="icon"
      style={{}}
    />
    <Text
      isActive={false}
      style={
        {
          "color": "#A0A0A0",
          "fontSize": 12,
          "fontWeight": "500",
        }
      }
    >
      Todos
    </Text>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 5,
        "height": 18,
        "maxWidth": 90,
        "opacity": 1,
      }
    }
    testID="button-PENDING"
  >
    <Image
      source="icon"
      style={{}}
    />
    <Text
      isActive={true}
      style={
        {
          "color": "#000000",
          "fontSize": 12,
          "fontWeight": "500",
        }
      }
    >
      Pendentes
    </Text>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 5,
        "height": 18,
        "maxWidth": 90,
        "opacity": 1,
      }
    }
    testID="button-PENDING"
  >
    <Image
      source="icon"
      style={{}}
    />
    <Text
      isActive={false}
      style={
        {
          "color": "#A0A0A0",
          "fontSize": 12,
          "fontWeight": "500",
        }
      }
    >
      Pendentes
    </Text>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 5,
        "height": 18,
        "maxWidth": 90,
        "opacity": 1,
      }
    }
    testID="button-PURCHASED"
  >
    <Image
      source="icon"
      style={{}}
    />
    <Text
      isActive={true}
      style={
        {
          "color": "#000000",
          "fontSize": 12,
          "fontWeight": "500",
        }
      }
    >
      Comprados
    </Text>
  </View>,
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "gap": 5,
        "height": 18,
        "maxWidth": 90,
        "opacity": 1,
      }
    }
    testID="button-PURCHASED"
  >
    <Image
      source="icon"
      style={{}}
    />
    <Text
      isActive={false}
      style={
        {
          "color": "#A0A0A0",
          "fontSize": 12,
          "fontWeight": "500",
        }
      }
    >
      Comprados
    </Text>
  </View>,
]
`;
